js/adapt-contrib-text.js

```diff
@@ -1,58 +1,55 @@
-define([
-  'core/js/adapt',
-  'core/js/views/componentView',
-  'core/js/models/componentModel'
-], function(Adapt, ComponentView, ComponentModel) {
+import Adapt from 'core/js/adapt';
+import ComponentView from 'core/js/views/componentView';
+import ComponentModel from 'core/js/models/componentModel';
 
-  class TextView extends ComponentView {
+class TextView extends ComponentView {
 
-    preRender() {
-      this.checkIfResetOnRevisit();
-    }
+  preRender() {
+    this.checkIfResetOnRevisit();
+  }
 
-    postRender() {
-      this.setReadyStatus();
+  postRender() {
+    this.setReadyStatus();
 
-      this.setupInview();
-    }
-
-    setupInview() {
-      const selector = this.getInviewElementSelector();
-      if (!selector) {
-        this.setCompletionStatus();
-        return;
-      }
+    this.setupInview();
+  }
 
-      this.setupInviewCompletion(selector);
+  setupInview() {
+    const selector = this.getInviewElementSelector();
+    if (!selector) {
+      this.setCompletionStatus();
+      return;
     }
 
-    /**
-     * determines which element should be used for inview logic - body, instruction or title - and returns the selector for that element
-     */
-    getInviewElementSelector() {
-      if (this.model.get('body')) return '.component__body';
+    this.setupInviewCompletion(selector);
+  }
 
-      if (this.model.get('instruction')) return '.component__instruction';
+  /**
+   * determines which element should be used for inview logic - body, instruction or title - and returns the selector for that element
+   */
+  getInviewElementSelector() {
+    if (this.model.get('body')) return '.component__body';
 
-      if (this.model.get('displayTitle')) return '.component__title';
+    if (this.model.get('instruction')) return '.component__instruction';
 
-      return null;
-    }
+    if (this.model.get('displayTitle')) return '.component__title';
+
+    return null;
+  }
 
-    checkIfResetOnRevisit() {
-      const isResetOnRevisit = this.model.get('_isResetOnRevisit');
+  checkIfResetOnRevisit() {
+    const isResetOnRevisit = this.model.get('_isResetOnRevisit');
 
-      // If reset is enabled set defaults
-      if (isResetOnRevisit) {
-        this.model.reset(isResetOnRevisit);
-      }
+    // If reset is enabled set defaults
+    if (isResetOnRevisit) {
+      this.model.reset(isResetOnRevisit);
     }
   }
+}
 
-  TextView.template = 'text';
+TextView.template = 'text.jsx';
 
-  return Adapt.register('text', {
-    model: ComponentModel.extend({}),// create a new class in the inheritance chain so it can be extended per component type if necessary later
-    view: TextView
-  });
+export default Adapt.register('text', {
+  model: ComponentModel,
```

Comment:
Extend from `ComponentModel` so that `TextModel` is it's own point in the hierarchy of inheritance.

Reply:
I extended this by creating a new TextModel class rather than using the extend() method. Thoughts?

Reply:
If it compiles and works I think it's fine, it's better than using the backbone syntax, but whichever way we do it, it needs to be uniform across the core. 

It almost looks as though it should be a separate file when written like that. I'm in two minds. A separate file would be a nice standard but it seems like overkill for a basic class definition. 

What do you think?

Reply:
I first did it as a separate file instinctively, then resolved into one to match existing components. I'm inclined to have a separate file to maintain separation of concerns very clearly (though I agree it looks a bit weird to have a file with an empty class def).

Reply:
I'm happy with a separate file.

Reply:
Done